File: src/js/fizzbuzz_recursive.js
(function(now,max,FizzBuzz){
  if(now%3 == 0 && now%5 == 0) FizzBuzz+='FizzBuzz';
  else if(now%5 == 0) FizzBuzz+='Buzz';
  else if(now%3 == 0) FizzBuzz+='Fizz';
  else FizzBuzz+=now;
  
  if(now>=max) {console.log(FizzBuzz);return;}
  FizzBuzz+=',';
  now++;
  arguments.callee(now,max,FizzBuzz);
})(1,100,'');

File: src/ruby/fizzbuzz_recursive.rb
def fizzbuzz(now,max,str)
  if(now%3 == 0 && now%5 == 0)
    str << 'FizzBuzz'
  elsif(now%5 == 0)
    str << 'Buzz'
  elsif(now%3 == 0)
    str << 'Fizz'
  else
    str << now.to_s
  end
  
  if(now >= max)
    puts str
    return
  end
  str << ','
  now += 1

  fizzbuzz(now,max,str)
end

fizzbuzz(1,100,'')
